core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -257,8 +257,9 @@ class LogCleaner(val config: CleanerConfig,
             cleanerManager.doneDeleting(topicPartition)
           }
       }
-      if (!cleaned)
-        backOffWaitLatch.await(config.backOffMs, TimeUnit.MILLISECONDS)
+      if (!cleaned) {
+        val _ = backOffWaitLatch.await(config.backOffMs, TimeUnit.MILLISECONDS)
```

Comment:
I see. Unfortunately, this is a bit of an implementation detail. The scala compiler could simply remove the local variable from the bytecode since it's not used (I think it does that with -optimise) and that would trigger the warning again. Maybe we want to disable the warning instead?